ci/test/actions-demo

```diff
@@ -38,9 +38,38 @@ cat <<EOF > main.workflow
 workflow "clone" {
   resolves = "test"
 }
+
+action "dependency four" {
+  uses = "docker://busybox"
+  runs = ["ls", "-ltr"]
+}
+
+action "dependency three" {
+  uses = "docker://busybox"
+  runs = ["ls", "-ltr"]
+}
+
+action "dependency two" {
+  needs = ["dependency three", "dependency four"]
+  uses = "docker://busybox"
+  runs = ["ls", "-ltr"]
+}
+
+action "dependency one" {
+  needs = ["dependency two"]
+  uses = "docker://busybox"
+  runs = ["ls", "-ltr"]
+}
+
 action "test" {
+  needs = ["dependency one"]
   uses = "https://gitlab.com/barbaragd/action-test@master"
 }
 EOF
 
 popper run
+
+# test --with-dependencies flag
+popper run test
+popper run --with-dependencies test
```

Comment:
can we check the output in order to ensure that only dependent actions get executed? Also, we can skip the previous `popper run test` as that functionality is being checked somewhere else

Reply:
@ivotron Are we required to match the output with a hardcoded output ? If thats the case, we might use `--dry-run` flag.

Reply:
yeah, using `--dry-run` would speedup the execution. No need to compare to hardcoded strings though. We could run with and without `--with-dependencies` and ensure that the latter is including dependent actions in the execution

Reply:
@ivotron On running without `--with-dependencies`, the actions defined in the `needs` attribute of the referenced action will get executed ?

Reply:
no, only the given action